subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/hived_algorithm.go

```diff
@@ -426,11 +450,17 @@ func (h *HivedAlgorithm) scheduleRegularAffinityGroup(sr schedulingRequest) (map
 			physicalPlacement[podGpuNum][i] = make(CellList, len(podGpus))
 			for j, gpu := range podGpus {
 				vGpu := gpu.(*VirtualCell)
+				if vGpu.GetPhysicalCell() != nil {
+					groupToPreempt := vGpu.GetPhysicalCell().GetAffinityGroup()
+					h.downgradeAffinityGroup(groupToPreempt)
+					klog.Infof("Affinity group %v downgraded to opportunistic due to intra-VC preemption",
+						groupToPreempt.name)
```

Comment:
Will we benefit from downgrade instead of immediately preempt? Because the vGpu is anyway need to be preempted.